src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+{
+    Commands::CopySceneResponse::Type response;
+
+    uint16_t endpoinTableSize = 0;
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, Attributes::SceneTableSize::Get(ctx.mRequestPath.mEndpointId, &endpoinTableSize)));
+
+    // Get Scene Table Instance
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable =
+        chip::scenes::GetSceneTableImpl(ctx.mRequestPath.mEndpointId, endpoinTableSize);
+
+    // Response data
+    response.groupIdentifierFrom = req.groupIdentifierFrom;
+    response.sceneIdentifierFrom = req.sceneIdentifierFrom;
+
+    // Verify Endpoint in group
+    VerifyOrReturn(nullptr != mGroupProvider);
+    if ((0 != req.groupIdentifierFrom &&
+         !mGroupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierFrom, mPath.mEndpointId)) ||
+        (0 != req.groupIdentifierTo &&
+         !mGroupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierTo, mPath.mEndpointId)))
+    {
+        response.status = to_underlying(Protocols::InteractionModel::Status::InvalidCommand);
+        ctx.mCommandHandler.AddResponse(mPath, response);
+        return;
+    }
+
+    uint8_t capacity = 0;
+    // Get Capacity
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->GetRemainingCapacity(ctx.mCommandHandler.GetAccessingFabricIndex(), capacity)));
+
+    // Checks if we copy a single scene or all of them
+    if (req.mode.GetField(app::Clusters::Scenes::ScenesCopyMode::kCopyAllScenes))
+    {
+        // Scene Table interface data
+        SceneId scenesInGroup[scenes::kMaxScenesPerFabric];
+        Span<SceneId> sceneList = Span<SceneId>(scenesInGroup);
+
+        // populate scene list
+        ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+            ctx, response,
+            sceneTable->GetAllSceneIdsInGroup(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierFrom, sceneList)));
+
+        if (0 == capacity)
+        {
+            response.status = to_underlying(Protocols::InteractionModel::Status::ResourceExhausted);
+            ctx.mCommandHandler.AddResponse(mPath, response);
+            return;
+        }
+
+        for (auto & sceneId : sceneList)
+        {
+            SceneTableEntry scene(SceneStorageId(sceneId, req.groupIdentifierFrom));
+            //  Insert in table
+            ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+                ctx, response,
+                sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene.mStorageId, scene)));
+
+            scene.mStorageId = SceneStorageId(sceneId, req.groupIdentifierTo);
+
+            ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+                ctx, response, sceneTable->SetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene)));
```

Comment:
Is this going to end up returning a RESOURCE_EXHAUSTED status if we run out of space?  I didn't catch how it does that.